aws-java-sdk-core/src/main/java/com/amazonaws/auth/InstanceProfileCredentialsProvider.java

```diff
@@ -76,7 +78,22 @@ public InstanceProfileCredentialsProvider() {
      *            true if credentials needs to be refreshed asynchronously else
      *            false.
      */
-    public InstanceProfileCredentialsProvider(boolean refreshCredentialsAsync) {
+    public InstanceProfileCredentialsProvider(boolean refreshCredentialsAsync) { this(refreshCredentialsAsync, true); }
+
+    /**
+     * Spins up a new thread to refresh the credentials asynchronously if
+     * refreshCredentialsAsync is set to true, otherwise the credentials will be
+     * refreshed from the instance metadata service synchronously,
+     *
+     * @param refreshCredentialsAsync
+     *            true if credentials needs to be refreshed asynchronously else
+     *            false.
+     * @param eagerlyRefreshCredentialsAsync
+     *            when set to false will not attempt to refresh credentials asynchronously
+     *            until after a call has been made to {@link #getCredentials()} - ensures that
+     *            {@link EC2CredentialsFetcher#getCredentials()} is only hit when this CredentialProvider is actually required
+     */
+    public InstanceProfileCredentialsProvider(boolean refreshCredentialsAsync, final boolean eagerlyRefreshCredentialsAsync) {
```

Comment:
This doesn't do what you think it does. By the time `getCredentialsCalled` is set to true in `getCredentials()`, this constructor has long since been run and executed thus won't execute again to pick up the change in `getCredentialsCalled`. It's effectively dead code.